plugins/search-react/src/context/SearchContext.tsx

```diff
@@ -50,12 +53,29 @@ export type SearchContextState = {
   pageCursor?: string;
 };
 
+// TODO(@backstage/techdocs-core): stop export this publicly when deprecating the context in @backstage/plugin-techdocs.
 /**
  * @public
  */
-export const SearchContext = createContext<SearchContextValue | undefined>(
-  undefined,
-);
+export const SearchContext = createVersionedContext<{
+  1: SearchContextValue;
+}>('search-context');
+
+/**
+ * @public
+ */
+export const useSearch = () => {
+  const context = useContext(SearchContext);
+  if (!context) {
+    throw new Error('useSearch must be used within a SearchContextProvider');
+  }
+
+  const value = context.atVersion(1);
+  if (!value) {
+    throw new Error('No context for version 1 found');
```

Comment:
Prolly mention `SearchContext v1` here instead, just to make it obvious what's going on if this error is encountered